src/layout/flexbox.js

```diff
@@ -0,0 +1,178 @@
+import computeLayout from 'css-layout';
+import d3 from 'd3';
+import innerDimensions from '../util/innerDimensions';
+
+function ownerSVGElement(node) {
+    while (node.ownerSVGElement) {
+        node = node.ownerSVGElement;
+    }
+    return node;
+}
+
+// parses the style attribute, converting it into a JavaScript object
+function parseStyle(style) {
+    if (!style) {
+        return {};
+    }
+    var properties = style.split(';');
+    var json = {};
+    properties.forEach(function(property) {
+        var components = property.split(':');
+        if (components.length === 2) {
+            var name = components[0].trim();
+            var value = components[1].trim();
+            json[name] = isNaN(value) ? value : Number(value);
+        }
+    });
+    return json;
+}
+
+// creates the structure required by the layout engine
+function createNodes(el) {
+    function getChildNodes() {
+        var children = [];
+        for (var i = 0; i < el.childNodes.length; i++) {
+            var child = el.childNodes[i];
+            if (child.nodeType === 1) {
+                if (child.getAttribute('layout-style')) {
+                    children.push(createNodes(child));
+                }
+            }
+        }
+        return children;
+    }
+    return {
+        style: parseStyle(el.getAttribute('layout-style')),
+        children: getChildNodes(el),
+        element: el
+    };
+}
+
+// takes the result of layout and applied it to the SVG elements
+function applyLayout(node, subtree) {
+    // don't set layout-width/height on layout root node
+    if (subtree) {
+        node.element.setAttribute('layout-width', node.layout.width);
+        node.element.setAttribute('layout-height', node.layout.height);
+    }
+
+    node.element.setAttribute('layout-x', node.layout.left);
+    node.element.setAttribute('layout-y', node.layout.top);
+
+    //for svg / rect set the position / dimensions via x/y/width/height properties
+    if (node.element.nodeName.match(/(?:svg|rect)/i)) {
+        node.element.setAttribute('width', node.layout.width);
+        node.element.setAttribute('height', node.layout.height);
+        node.element.setAttribute('x', node.layout.left);
+        node.element.setAttribute('y', node.layout.top);
+    } else if (subtree) {
```

Comment:
I was just thinking that the logic to apply to an `svg`/`rect` should be the same as the others. Maybe that just means we should exclude `x` and `y`?